db/migrate/20231121184619_add_confirmable_to_devise.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class AddConfirmableToDevise < ActiveRecord::Migration[7.1]
+  def up
+    change_table :users, bulk: true do |t|
+      t.string :confirmation_token
+      t.datetime :confirmed_at
+      t.datetime :confirmation_sent_at
+      t.string :unconfirmed_email
+      t.index :confirmation_token, unique: true
+    end
+  end
+
+  def down
+    change_table :users, bulk: true do |t|
+      t.remove :confirmation_token
+      t.remove :confirmed_at
+      t.remove :confirmation_sent_at
+      t.remove :unconfirmed_email
+    end
+  end
```

Comment:
I believe there's no need to specify up/down for this migration. If you just use `def change` with the "up" part, rails will know how to rollback the migration.